apps/ftu/test/unit/wifi_test.js

```diff
@@ -210,72 +153,411 @@ suite('wifi > ', function() {
     test('should execute callback when scan fails', function() {
       var fakeCallback = this.sinon.spy();
 
-      var stub = this.sinon.stub(MockNavigatorMozWifiManager, 'getNetworks');
-      stub.onFirstCall().returns(
-        {
+      this.sinon.stub(MockNavigatorMozWifiManager, 'getNetworks')
+        .onFirstCall().returns({
           set onerror(callback) {
             this.error = {
               name: 'error'
             };
             callback();
           }
-        }
-      );
+        });
 
       WifiManager.scan(fakeCallback);
       assert.isTrue(fakeCallback.calledOnce);
     });
 
     test('timeout error', function(done) {
-      var consoleSpy = this.sinon.spy(console, 'warn');
-      var stub = this.sinon.stub(MockNavigatorMozWifiManager, 'getNetworks',
-        function() {
-          return {};
-      });
-      WifiManager.scan(function(networks) {
-        assert.ok(showOverlayStub.called, 'shows loading overlay');
+      this.sinon.stub(MockNavigatorMozWifiManager, 'getNetworks').returns({});
+      this.sinon.spy(console, 'warn');
+
+      WifiManager.scan(networks => {
+        assert.ok(utils.overlay.show.called, 'shows loading overlay');
         assert.isUndefined(networks, 'no networks returned');
-        assert.ok(consoleSpy.calledOnce);
+        assert.ok(console.warn.called);
         done();
       });
+
       clock.tick(10000);
     });
   });
 
-  suite('Join networks', function() {
-    var connectStub;
+  suite('Choose a network >', function() {
+    var network,
+        networkDOM,
+        fakeEvent;
+
     setup(function() {
-      connectStub = this.sinon.stub(WifiUI, 'connect',
-        function(ssid, password, user) {
-          return;
-      });
-      createDOM();
+      WifiManager.networks = fakeNetworks;
+      WifiUI.renderNetworks(fakeNetworks);
+      UIManager.activationScreen =
+        document.getElementById('activation-screen');
+      UIManager.mainTitle = document.getElementById('main-title');
+      UIManager.wifiJoinButton = document.getElementById('wifi-join-button');
+      UIManager.navBar = document.getElementById('nav-bar');
+
+      network = fakeNetworks[0];
+      networkDOM = document.querySelector('li[data-ssid="' +
+                                          network.ssid + '"]');
+      fakeEvent = {
+        target: networkDOM
+      };
     });
 
-    test('Should join to a wifi network', function() {
-      var password =
-        document.getElementById('wifi_password');
-      var ssid = document.getElementById('wifi_ssid');
-      var user = document.getElementById('wifi_user');
+    teardown(function() {
+      UIManager.activationScreen = null;
+      UIManager.mainTitle = null;
+      UIManager.wifiJoinButton = null;
+      UIManager.navBar = null;
+    });
 
-      ssid.value = 'testSSID';
-      password.value = 'testPassword';
+    test('Open network', function() {
+      this.sinon.stub(WifiManager, 'getNetwork', function get(id) {
```

Comment:
need to stub this way instead of using `return Promise.resolve()`to avoid any use of `setTimeout` on tests.
If use the latter, the resolve will be executed on the next cycle and the test would be finished before that.